### PR TITLE
Wallet HD refactor

### LINKED ISSUE #29: [SDK][Wallet] Confidential support
At the moment we cannot handle confidential transactions properly. This means the Wallet instance should have knowledge of blinding keys

### LINKED ISSUE #30: [SDK][coinselect] Support unblinding of utxos
At the moment we do not support confidential utxos, we need to support the unblinding of unspents via given blinding key.

### LINKED ISSUE #34: [SDK][Wallet] HD support
The signing process is delegated to the Identity class. The Wallet then becomes a stateless PSET controller (create, update, extract, finalize) and coin controller (unblind && fetch transactions)

This means we need to initialize our wallet with a list of addresses and blinding keys. This class will be used by higher componenents combined with the Identity abstraction to sing a PSET (created with the help of the wallet).

### PR DESCRIPTION
**Pull request by tiero**
This commit starts updating the wallet to be stateless and delegating to upper components (ie. Identity) the duty to sign. In order to initialize the Wallet instance we need an array of `AddressInterface`. Those addresses will be used for coinselection.

~~The updateTx is not working yet, needs to fetch proofs and make unblinding of alla unspents and coin selection to properly add inputs.~~ 
UPDATE: Now it works. It always use the first address of the internal cache, should be changed once the Identity will get getNextAddress and getNextChangeAddress methods

It fixes #30 
It fixes #34 
It fixes #29 

Please @louisinger review this

### CODE REVIEW

File: src/trade.ts [hunk 1]
```diff
@@ -195,32 +196,43 @@ export class Trade extends Core implements TradeInterface {
...
-        );
-      })
+        )
+      )
     );
 
-    // TODO this will need to call the identity class to get the next address, for now we hardcode the first one from cache.
-    const firstAddress = wallet.addresses[0].confidentialAddress;
+    let traderUnblindedUtxos: UtxoInterface[] = [];
+    arrayOfArrayOfUtxos.forEach((a: UtxoInterface[]) => {
```
Comment (louisinger) [inline src/trade.ts]: You can use the `flat()` method maybe

File: src/wallet.ts [hunk 2]
```diff
@@ -1,230 +1,281 @@
...
+import { fromAssetHash, toAssetHash } from './utils';
+import { AddressInterface } from './identity';
+
+/**
+ * Wallet abstraction.
+ */
+export interface WalletInterface {
   network: Network;
+  addresses: AddressInterface[];
+  blindingByScript: Record<string, Buffer>;
```
Comment (louisinger) [inline src/wallet.ts:20]: could be renamed such as `blindingPrivateKeyByScript`

File: src/wallet.ts [hunk 3]
```diff
@@ -1,230 +1,281 @@
...
+ * @method createTx init empty PSET.
+ * @method updateTx update a PSET with outputs and inputs (for Swap tx).
+ */
+export class Wallet implements WalletInterface {
   network: Network;
-  address: string;
-  script: string;
-  constructor({ address, network }: { address: string; network: Network }) {
-    const payment = payments.p2wpkh({ address, network });
+  addresses: AddressInterface[] = [];
```
Comment (louisinger) [inline src/wallet.ts:44]: the member `addresses` is not used: only for creating blindingKeyByScript in the constructor.

Comment (tiero) [reply src/wallet.ts:44]: Yes it is externally, in marketOrderRequest for example to retrieve the addresses to fetch utxos from

**Pull request merged**